### PR TITLE
quicktype 6.0.0 (new formula)

### PR DESCRIPTION
**Pull request by dvdsgl**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/quicktype.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system "#{bin}/quicktype", "-o", "LatestBlock.cs", "https://blockchain.info/latestblock"
```
Comment (fxcoudert) [inline Formula/quicktype.rb]: Can we check that the LastestBlock.cs file has indeed been written? Or better, match some known content in it?

File: Formula/quicktype.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require "language/node"
+
+class Quicktype < Formula
+  desc "Generate types and converters from JSON, Schema, and GraphQL"
+  homepage "https://github.com/quicktype/quicktype#readme"
```
Comment (fxcoudert) [inline Formula/quicktype.rb]: Please remove the `#readme`

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/quicktype.rb (hunk 1 above)
Comment (dvdsgl) [reply Formula/quicktype.rb]: May I install test-only dependencies? quicktype generates code, so if I could install a compiler, I can run a comprehensive test.

Comment (fxcoudert) [reply Formula/quicktype.rb]: No test-only deps. But a good intermediate would be to grep for some known pattern in the generated code, for example.

**Comment by dvdsgl** [comment]
* Fixed link as requested
* Added an offline test that checks for valid tool output

File: Formula/quicktype.rb (hunk 2 above)
Comment (dvdsgl) [reply Formula/quicktype.rb]: Removed.

File: Formula/quicktype.rb (hunk 1 above)
Comment (dvdsgl) [reply Formula/quicktype.rb]: Done.

File: Formula/quicktype.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    (testpath / "sample.json").write <<~EOS
```
Comment (DomT4) [inline Formula/quicktype.rb]: The spacing is redundant here. `(testpath/"sample.json").write <<~EOS` will work fine.

**Comment by DomT4** [comment]
> C: 18: col 1: Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<~`.

Ugh. The wording of that error message remains 🗑. Keep meaning to look into where it's coming from.

File: Formula/quicktype.rb (hunk 3 above)
Comment (dvdsgl) [reply Formula/quicktype.rb]: Fixed.

**Comment by dvdsgl** [comment]
> C: 18: col 1: Use 2 spaces for indentation in a heredoc by using <<~ instead of <<~.

Is there something I can do to fix this issue? I don't understand what's wrong.

**Comment by fxcoudert** [comment]
I think it's your braces {} at lines 18 and 21 which should be indented 2 spaces right.

**Comment by DomT4** [comment]
Yeah, @fxcoudert has it right. Essentially the error message is saying _"Hey you need to use a two-space indent inside an EOS block"_ but instead comes out with that phrasing which so far has thoroughly confused at least a handful of people.

**Comment by dvdsgl** [comment]
Alright, indentation and `assert` vs `assert_match` errors fixed.

**Review by fxcoudert — APPROVED** [review]

**Pull request closed without merging**